packages/xo-web/src/common/editable/index.js

```diff
@@ -388,10 +410,34 @@ export class Select extends Editable {
     return children || <span>{renderer ? renderer(value) : value}</span>
   }
 
+  _selectAll = () => {
+    const { onChange, options } = this.props
+    return onChange(options)
+  }
+
   _renderEdition () {
-    const { saving, valueKey } = this.state
-    const { options } = this.props
+    const { hasSelectAll, multi, options } = this.props
+    if (multi || hasSelectAll) {
+      return (
+        <form
+          className='input-group'
+          ref={form => {
+            form && this.setState({ form })
```

Comment:
Why do you need to add it to the `state`?

Reply:
to add event `focusout`  in `componentDidUpdate`  (when we display `form` to edit `CPU mask`)

Reply:
`onBlur={...}`?